vllm/v1/worker/neuron_worker.py

```diff
@@ -0,0 +1,189 @@
+
+        if self.model_config.trust_remote_code:
+            # note: lazy import to avoid importing torch before initializing
+            from vllm.utils import init_cached_hf_modules
+            init_cached_hf_modules()
+
+        self.profiler = None
+
+        if self.model_config.seed is None:
+            self.model_config.seed = 0
+
+    @torch.inference_mode()
+    def determine_available_memory(self) -> int:
+        self.model_runner.profile_run()
+        for _ in range(10):
+            try:
+                memory_usage, total_memory = get_current_memory_usage(
+                    self.rank)
+                break
+            except json.JSONDecodeError:
+                continue
+        kv_cache_memory_available = (total_memory *
+                                     self.cache_config.gpu_memory_utilization)
+        return int(kv_cache_memory_available - memory_usage)
+
+    def init_device(self):
+        if self.device_config.device.type == "cpu":
+
+            # Initialize the distributed environment.
+            init_worker_distributed_environment(self.parallel_config,
+                                                self.rank,
+                                                self.distributed_init_method,
+                                                self.local_rank)
+
+            self.device = xm.xla_device()
+        else:
+            raise RuntimeError(
+                f"Not support device type: {self.device_config.device}")
+
+        # Set random seed.
+        set_random_seed(self.model_config.seed)
+
+        # Construct the model runner
+        with torch.inference_mode():
+            self.model_runner: NeuronModelRunner = NeuronModelRunner(
+                self.vllm_config, self.device)
+
+    def execute_model(
+        self,
+        scheduler_output: "SchedulerOutput",
+    ) -> Optional[ModelRunnerOutput]:
+        output = self.model_runner.execute_model(scheduler_output)
+        return output if self.is_driver_worker else None
+
+    def profile(self, is_start: bool = True):
+        raise NotImplementedError("Profiling is not implemented.")
+
+    def load_model(self) -> None:
+        self.model_runner.load_model()
+
+    def compile_or_warm_up_model(self):
+        # TODO: Implement AOT compilation logic here...
+        self.model_runner.capture_model()
+
+    def get_model(self) -> nn.Module:
+        return self.model_runner.get_model()
+
+    def get_kv_cache_spec(self) -> KVCacheSpec:
+        return self.model_runner.get_kv_cache_spec()
+
+    def initialize_cache(self, kv_cache_config: KVCacheConfig) -> None:
+        # TODO(gnovack) - validate num_device_blocks
+        self.model_runner.initialize_kv_cache(kv_cache_config)
+
+    def initialize_from_config(self, kv_cache_config: KVCacheConfig) -> None:
+        """Allocate device KV cache with the specified kv_cache_config."""
+        self.model_runner.initialize_kv_cache(kv_cache_config)
```

Comment:
do we need both of these functions? seems they both do the same thing. I'm guessing only one is called upstream at this point

Reply:
removed `initialize_cache` (V0) and kept `initialize_from_config` (V1)